src/session/Session.ts

```diff
@@ -177,21 +180,38 @@ export class Session extends OTEventEmitter<{
       return localPublisher;
     }
 
-    window.call.on("participant-joined", (dailyEvent) => {
-      if (!dailyEvent) {
-        console.debug("No Daily event");
-        return;
-      }
-      this.eventHandler.onParticipantJoined(dailyEvent.participant);
-    });
+    window.call
+      .on("participant-joined", (dailyEvent) => {
+        // remote
+        if (!dailyEvent) {
+          console.debug("No Daily event");
+          return;
+        }
+        this.eventHandler.onParticipantJoined(dailyEvent.participant);
+      })
+      .once("participant-updated", (dailyEvent) => {
+        // run once for local participant only
+        if (!dailyEvent) {
```

Comment:
Wondering about the same thing here as the publisher above: how do we know this will get fired and get fired first for the local participant? What happens if it does not get fired or if our one chance to process it is actually for a remote participant? Will the `"streamCreated"` event never be emitted then? Is that recoverable?